src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -81,7 +81,8 @@ public async Task<LocationOrLocationLinks> Handle(DefinitionParams request, Canc
                 return null;
             }
 
-            var originTagDescriptor = originTagHelperBinding.Descriptors.SingleOrDefault();
+            // We use EndsWith as the Descriptor.Name is fully qualified but the TagName doesn't have to be
+            var originTagDescriptor = originTagHelperBinding.Descriptors.FirstOrDefault(d => d.Name.EndsWith(originTagHelperBinding.TagName));
```

Comment:
Instead of doing a name check it might be better to either remove all ["attribute" bound](https://github.com/dotnet/aspnetcore/blob/d88935709b8908f371aa97e32a3ce4a74af2368f/src/Razor/Microsoft.AspNetCore.Razor.Language/src/TagHelperBinding.cs#L42-L43) descriptors or re-query some pieces through our [TagHelperMathcingConventions](https://github.com/dotnet/aspnetcore/blob/d88935709b8908f371aa97e32a3ce4a74af2368f/src/Razor/Microsoft.AspNetCore.Razor.Language/src/TagHelperMatchingConventions.cs)

Reply:
Updated!